proto/asn1/asn1-universal-types/asn1_universal_types.proto

```diff
@@ -0,0 +1,77 @@
+// This proto represents the Universal types
+// found in X509 Certifiactes.
+syntax = "proto2";
+
+package asn1_universal_types;
+
+// See X.690 (2015), 8.1.2.5.
+enum Encoding {
+  Primitive = 0;
+  Constructed = 1;
+}
+
+// See X.690 (2015), 8.6.
+message BitString {
+  // BitString is either constructed or primitive (X.690 (2015), 8.6.1).
+  required Encoding encoding = 1;
+  required bytes val = 2;
+}
+
+// See X.690 (2015), 8.3.
+message Integer {
+  required bytes val = 2;
+}
+
+// See X.690 (2015), 11.8.
+message UTCTime {
+  required TimeValue year_tens_digit = 1;
+  required TimeValue year_ones_digit = 2;
+  required TimeValue month_tens_digit = 3;
+  required TimeValue month_ones_digit = 4;
+  required TimeValue day_tens_digit = 5;
+  required TimeValue day_ones_digit = 6;
+  required TimeValue hour_tens_digit = 7;
+  required TimeValue hour_ones_digit = 8;
+  required TimeValue minute_tens_digit = 9;
+  required TimeValue minute_ones_digit = 10;
+  required TimeValue second_tens_digit = 11;
+  required TimeValue second_ones_digit = 12;
+  // See X.690 (2015), 11.8.1: Encoding terminates with "Z".
+  required bool zulu = 13;
```

Comment:
This is where we'll want to discuss invalid DER or not, since you're allowing for invalid DER here, as well as in situations like where the minutes exceeds 60 minutes or the hours exceeds 24 hours

Reply:
We've agreed that we want to only use valid dates and have ensured that in the code.